core/indigo-core/common/base_cpp/array.h

```diff
@@ -25,6 +25,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <utility>
+#include <list>
```

Comment:
IMHO, this change can confuse further refactorings in perspective of removing `array.h`. Better insert `#include <list>` where the list is used.